### PR TITLE
models: use correct type definitions

### PR DESCRIPTION
**Pull request by kjdelisle**
connected to https://github.com/strongloop/loopback-datasource-juggler/issues/1434

This should fix the test failures seen in microgateway, related to loopback-datasource-juggler@2.55.1

### CODE REVIEW

**Comment by kjdelisle** [comment]
Also, note that the Travis CI failures are due to this repo not having a `.travis.yml` file.

File: common/models/product.json [hunk 1]
```diff
@@ -19,8 +19,7 @@
       "url": "string"
     },
     "document": {
-      "apis": 
-        {},
+      "apis": "object",
```
Comment (rmg) [inline common/models/product.json]: I notice that this isn't being done _everywhere_ - does that mean it isn't necessary? Or that the spots that were missed are dead code?

**Comment by rmg** [comment]
Would you mind adding a trivial `.travis.yml` file? Even just verifying the deps are all installable from public npm is useful since there's always the potential to accidentally add a dependency on an unpublished version.

This should be sufficient:
```yaml
sudo: false
language: node_js
node_js:
 - "4"
 - "6"
 - "8"
```

File: common/models/product.json (hunk 1 above)
Comment (jgeddes2001) [inline common/models/product.json]: I think this is related to a change in behavior and wasn't required before..

Comment (kjdelisle) [inline common/models/product.json]: It was my understanding that if you wanted to allow _any_ object there, that you'd use `"object"`, but if you wanted to explicitly define the properties that make up that object, you'd have to have at least one key-value pair in the definition.

Comment (kjdelisle) [inline common/models/product.json]: The docs aren't 100% clear in either direction, though, so I'm not sure. @ritch would not supporting `"someProperty": { }` be considered a breaking API change, or was it never technically allowed?

**Comment by kjdelisle** [comment]
#12 has landed, and this is rebased. If all tests pass, I'll merge.

**Pull request merged**